### PR TITLE
Expand Templates

### PR DESCRIPTION
**Pull request by jdunkerley**
### Pull Request Description

- Update all the templates to the newest versions.
- Fix any defects in the templates.
- Update the rust and scala code to package and download the templates.
- Update the front end to have images for every template (with some Gemini assists).

https://github.com/user-attachments/assets/59a1f845-f409-45bb-9c77-fe3a2d1262b1

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

### CODE REVIEW

**Review by kazcw — APPROVED** [review]
Approving build script changes

File: app/dashboard/src/layouts/Samples.tsx [hunk 1]
```diff
@@ -130,23 +177,6 @@ function ProjectTile(props: InternalProjectTileProps) {
       {/* Although this component is instantiated multiple times, it has a unique role and hence
        * its own opacity. */}
       {/* eslint-disable-next-line no-restricted-syntax */}
```
Comment (somebody1234) [inline app/dashboard/src/layouts/Samples.tsx]: this comment and the one directly above are no longer necessary

**Review by somebody1234 — APPROVED** [review]
CR :white_check_mark: for dashboard code

**Review by MrFlashAccount — APPROVED** [review]
LGTM, but please don't merge it, I'll push a few improvements tomorrow

**Review by 4e6 — APPROVED** [review]

**Review by Frizi — APPROVED** [review]

**Review by AdRiley — APPROVED** [review]

**Review by JaroslavTulach — APPROVED** [review]

**Comment by MrFlashAccount** [comment]
https://github.com/user-attachments/assets/6494379b-57c6-4b0b-9519-8648434788ab

Did some small improvements

**Pull request merged**